addons/sale_coupon/models/sale_order.py

```diff
@@ -154,10 +154,13 @@ def _get_reward_values_fixed_amount(self, program):
         # This allow manual overwrite of taxes for promotion.
         if program.discount_line_product_id.taxes_id:
             line_taxes = self.fiscal_position_id.map_tax(program.discount_line_product_id.taxes_id) if self.fiscal_position_id else program.discount_line_product_id.taxes_id
+            lines = self._get_paid_order_lines().filtered('price_reduce')
+            lines_price = sum(lines.mapped(lambda l: l.price_reduce * l.product_uom_qty))
+            discount_amount_without_tax = min(lines_price, discount_amount)
```

Comment:
It's a detail but there is no need for another variable
```suggestion
            discount_amount = min(sum(lines.mapped(lambda l: l.price_reduce * l.product_uom_qty)), discount_amount)
```